gammapy/datasets/core.py

```diff
@@ -229,9 +230,16 @@ def stat_sum(self):
         """Compute joint statistic function value."""
         stat_sum = 0
         # TODO: add parallel evaluation of likelihoods
+        prior_stat_sum = 0
         for dataset in self:
             stat_sum += dataset.stat_sum()
-        return stat_sum
+            # remove prior_fit_statistic from individual dataset to avoid double counting
+            if dataset.models is not None:
+                prior_stat_sum -= prior_fit_statistic(dataset.models.priors)
+        # compute prior_fit_statistics from all datasets
+        if self.models is not None:
+            prior_stat_sum += prior_fit_statistic(self.models.priors)
```

Comment:
Why does one prior count positive the other negative?

Reply:
`Datasets.model` includes all models, why is there an independent sum for `dataset.models`?